apis/keda/v1alpha1/scaledobject_types.go

```diff
@@ -254,6 +259,25 @@ func (so *ScaledObject) GetHPAMaxReplicas() int32 {
 	return defaultHPAMaxReplicas
 }
 
+// CheckScaleTargetRefIfExist checks if scaleTargetRef of ScaledObject exists
+func (so *ScaledObject) CheckScaleTargetRefIfExist(ctx context.Context) error {
+	soGvkr, err := ParseGVKR(restMapper, so.Spec.ScaleTargetRef.APIVersion, so.Spec.ScaleTargetRef.Kind)
+	if err != nil {
+		msg := "Failed to parse Group, Version, Kind, Resource"
+		scaledobjectlog.Error(err, msg, "apiVersion", so.Spec.ScaleTargetRef.APIVersion, "kind", so.Spec.ScaleTargetRef.Kind)
+		return err
+	}
+	gvkString := soGvkr.GVKString()
+	unstruct := &unstructured.Unstructured{}
+	unstruct.SetGroupVersionKind(soGvkr.GroupVersionKind())
+	if err := kc.Get(ctx, client.ObjectKey{Namespace: so.Namespace, Name: so.Spec.ScaleTargetRef.Name}, unstruct); err != nil {
+		// resource doesn't exist
+		scaledobjectlog.Error(err, message.ScaleTargetNotFoundMsg, "resource", gvkString, "name", so.Spec.ScaleTargetRef.Name)
+		return err
+	}
+	return nil
```

Comment:
As this can potentially happen before the targetRef is available through the cached client it should check the item using `getFromCacheOrDirect` to rely on user configuration to try using the direct client

Reply:
> As this can potentially happen before the targetRef is available through the cached client it should check the item using `getFromCacheOrDirect` to rely on user configuration to try using the direct client

Thank you for your comment. I have follow your idea and change the code already. Hope for your reply. MuchThanks.

Reply:
> As this can potentially happen before the targetRef is available through the cached client it should check the item using `getFromCacheOrDirect` to rely on user configuration to try using the direct client

Thank you for your comment. I have follow your idea and change the code already. Hope for your reply soon. MuchThanks.